src/routes/(app)/settings/+page.server.ts

```diff
@@ -1,36 +1,30 @@
 import type { Actions, PageServerLoad } from "./$types";
 import { fail } from "@sveltejs/kit";
-import { authorize } from "$lib/utils/authorization";
-import apiNames from "$lib/utils/apiNames";
+import { redirect } from "$lib/utils/redirect";
 
 export const load: PageServerLoad = async ({ locals }) => {
   const { user, prisma } = locals;
-  authorize(apiNames.MEMBER.UPDATE, user);
-
-  const subscriptionSettings = user
-    ? await prisma.subscriptionSetting.findMany({
-        where: {
-          memberId: user.memberId,
-        },
-      })
-    : [];
+  if (user?.memberId == null) throw redirect(302, "/home");
 
+  const subscriptionSettings = await prisma.subscriptionSetting.findMany({
+    where: {
+      memberId: user.memberId,
+    },
+  });
```

Comment:
If a user is not logged in, instead of redirecting them we should show an error to be clearer. This redirect doesn't make much sense. `throw error(401, "Not logged in")` instead (can't remember exact syntax, check other `page.server.ts` files or even the `authorize` function